common/lib/xmodule/xmodule/course_module.py

```diff
@@ -269,6 +269,44 @@ def get_available_providers():
     return available_providers
 
 
+class DiscussionTopicMapping(Dict):
+    """
+    DiscussionTopicMapping field, which includes validation of the topic id
+    """
+
+    _default = {}
+
+    def from_json(self, value):
```

Comment:
Can we give this a more descriptive name? The name here doesn't really express what the comment says the function is doing.

Reply:
All custom data types for advanced settings seem to need this function. I think its called by code whenever data needs to be read/handled